lib/thredded/per_notifier_pref.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+module Thredded
+  # in this class we dynamically set up an accessor for all the Thredded.notifiers (by their key)
+  # as we don't know in advance what the keys are,
+  # we try to not pollute the namespace
+  # nonetheless some keys would be bad, so we sanity check those on setup
+  class PerNotifierPref
```

Comment:
@glebm yes for a plugin in general, but think that specifically notifiers could have a fairly limited and therefore easy-to-constrain role. 

I think we can limit complexity on this with the solution you've outlined, and still allow for easy creation of new notifiers.

(I think it would be good for Notifier authors not to have to worry too much about db. Yes, if you're providing a Pushover/Slack notifer you have to provide the "user key" (might be email address, might be a separate user-key) somewhere, but I think it makes sense that you don't have to then add the preferences, which could well be standard.)